collector/lib/CollectorConfig.cpp

```diff
@@ -422,33 +425,120 @@ void CollectorConfig::YamlConfigToConfig(YAML::Node& yamlConfig) {
   externalIpsConfig->set_enable(enableExternalIps);
 
   SetRuntimeConfig(runtime_config);
-  CLOG(INFO) << "Runtime configuration:";
-  CLOG(INFO) << GetRuntimeConfigStr();
+  CLOG(INFO) << "Runtime configuration:\n"
+             << GetRuntimeConfigStr();
 
   return;
 }
 
-void CollectorConfig::HandleConfig(const std::filesystem::path& filePath) {
+bool CollectorConfig::HandleConfig(const std::filesystem::path& filePath) {
   if (!std::filesystem::exists(filePath)) {
     CLOG(DEBUG) << "No configuration file found. " << filePath;
-    return;
+    return true;
   }
 
   try {
     YAML::Node yamlConfig = YAML::LoadFile(filePath);
     YamlConfigToConfig(yamlConfig);
   } catch (const YAML::BadFile& e) {
-    CLOG(FATAL) << "Failed to open the configuration file: " << filePath
-                << ". Error: " << e.what();
+    CLOG(ERROR) << "Failed to open the configuration file: " << filePath << ". Error: " << e.what();
+    return false;
   } catch (const YAML::ParserException& e) {
-    CLOG(FATAL) << "Failed to parse the configuration file: " << filePath
-                << ". Error: " << e.what();
+    CLOG(ERROR) << "Failed to parse the configuration file: " << filePath << ". Error: " << e.what();
+    return false;
   } catch (const YAML::Exception& e) {
-    CLOG(FATAL) << "An error occurred while loading the configuration file: " << filePath
-                << ". Error: " << e.what();
+    CLOG(ERROR) << "An error occurred while loading the configuration file: " << filePath << ". Error: " << e.what();
+    return false;
   } catch (const std::exception& e) {
-    CLOG(FATAL) << "An unexpected error occurred while trying to read: " << filePath << e.what();
+    CLOG(ERROR) << "An unexpected error occurred while trying to read: " << filePath << e.what();
+    return false;
+  }
+
+  return true;
+}
+
+void CollectorConfig::WaitForFileToExist(const std::filesystem::path& filePath) {
+  int count = 0;
+  while (!std::filesystem::exists(filePath) && !thread_.should_stop()) {
+    sleep(1);
+    count++;
+    if (count > 45) {
+      std::unique_lock<std::shared_mutex> lock(mutex_);
+      runtime_config_.reset();
+    }
+  }
+}
+
+int CollectorConfig::WaitForInotifyAddWatch(int fd, const std::filesystem::path& filePath) {
+  while (!thread_.should_stop()) {
```

Comment:
Now I'm thinking, why does this methid need a loop? If we already waited for the file to exist, is not like the OS is goning to forbid adding the watch now and randomly decide to allow it the next second, I think we should be able to just remove this:
```suggestion
```